### ISSUE TITLE
External CA mode fail during kubelet-finalize

### ISSUE THREAD

**Issue by berthoug** [kubernetes/kubernetes#89824]
**What happened**:
I set up the external CA following the instruction in https://kubernetes.io/docs/setup/best-practices/certificates/
I put the ca.crt in /etc/kubernetes/pki, created the different certificates and configurations (as indicated [here](https://kubernetes.io/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts)
When I run kubeadm init it fails with the following message:
`[kubelet-finalize] Updating "/etc/kubernetes/kubelet.conf" to point to a rotatable kubelet client certificate and key
error execution phase kubelet-finalize/experimental-cert-rotation: the file "/etc/kubernetes/kubelet.conf" does not contain authentication for user "ip-172-31-20-205"
To see the stack trace of this error execute with --v=5 or higher`
 
**What you expected to happen**:
As I followed the instruction of the documentation to create the `/etc/kubernetes/kubelet.conf` file I would have expected it to be set up properly for kubadm init to work.

I tried to set `rotateCertificates: false` in the `KubeletConfiguration` part of my `kubeadm.conf` but it seems to be ignored (If I understand [the code](https://github.com/kubernetes/kubernetes/blob/ff809a5d953ba778270ce8790b21d394821e1e28/cmd/kubeadm/app/componentconfigs/kubelet.go#L173) correctly it is hardcoded to true).

**How to reproduce it (as minimally and precisely as possible)**:
put the ca.crt in /etc/kubernetes/pki, created the different certificates and configurations (as indicated [here](https://kubernetes.io/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts)
Run kubadm init

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.18
- Cloud provider or hardware configuration: m5.2xlarge
- OS (e.g: `cat /etc/os-release`): 
`NAME="Ubuntu"
VERSION="18.04.3 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.3 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
`
- Kernel (e.g. `uname -a`): `Linux ip-172-31-20-205 4.15.0-1057-aws #59-Ubuntu SMP Wed Dec 4 10:02:00 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
`
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by neolit123** [comment]
> I tried to set rotateCertificates: false in the KubeletConfiguration part of my kubeadm.conf but it seems to be ignored (If I understand the code correctly it is hardcoded to true).

yes it is, because there is no way for us to tell if the user set the value to false or the defaulting of go structs set the value of false. if the KubeletConfiguration had `*bool` it would have been possible.

> execution phase kubelet-finalize/experimental-cert-rotation

you can skip this phase with `--skip-phases=kubelet-finalize/experimental-cert-rotation` but before we do that i would like understand more about the issue.

> https://kubernetes.io/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts

this guide there is valid. it makes you use files instead of hardcoded cert/key.

> does not contain authentication for user "ip-172-31-20-205" To see the stack trace of this error execute with --v=5 or higher

can  you show the stack trace?
also can you show your `kubelet.conf` while replacing the sensitive data with comments.

basically this phase's responsibility is to make sure that your kubelet.conf does not include hardcoded certificate/key for this node and instead point to files that are rotated in /var/lib/kubelet/pki...

**Comment by berthoug** [comment]
Hi,

Thanks for you quick answer.
I think it is mostly a documentation problem. Looking more into the kubernetes code I figured out that the  `<credential-name>` point 2 of the [documentation](https://kubernetes.io/docs/setup/best-practices/certificates/#configure-certificates-for-user-accounts) should be set to `system:node:ip-172-31-20-205` which is indicated as the `default CN` in the table above while credential name says `default-auth`.

**Comment by neolit123** [comment]
your observation that passing `system:node:ip-172-31-20-205` to `kubectl config set-credentials` is valid. this is not a core kubernetes/kubernetes issue though.

please file this same ticket in the kubernetes/website issue tracker with an explanation. even better, send a PR to fix the issue if you can.

thanks
/kind documentation
/remove-area kubeadm
/sig auth
/remove-sig cluster-lifecycle
/close

**Comment by neolit123** [comment]
please, ping me on the new issue / PR.

**Comment by vrutkovs** [comment]
Filed https://github.com/kubernetes/website/issues/44876
